### PR TITLE
FS-21: setup CI/CD

### PR DESCRIPTION
**Pull request by gurdeep111994**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - name: test
+        run: npm run test
+      - name: test coverage
+        run: npm run test:cov
+      
+      - uses: akhileshns/heroku-deploy@v3.12.12
+        with:
+          heroku_api_key: ${{ secrets.HEROKUNEXTJSDEPLOYMENT }}
+          heroku_app_name: food-secure-back-end
+          heroku_email: "gs22111994@gmail.com"
```
Comment (chirgjn) [inline .github/workflows/main.yml]: These should also be moved to environment variables

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: pull from Heroku
+        run: git pull --rebase heroku master
```
Comment (chirgjn) [inline .github/workflows/main.yml]: Why is this step needed?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - name: lint
+        run: npm run lint
+      - name: build
+        run: npm run build
+      - name: test
+        run: npm run test
+      - name: test coverage
+        run: npm run test:cov
+      
+      - uses: akhileshns/heroku-deploy@v3.12.12
```
Comment (chirgjn) [inline .github/workflows/main.yml]: Can we avoid using this action, I think we discussed this last time.

**Review by chirgjn — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (gurdeep111994) [reply .github/workflows/main.yml]: yes, I tried to remove it but getting errors in deployment. I'll check

File: .github/workflows/main.yml (hunk 2 above)
Comment (gurdeep111994) [reply .github/workflows/main.yml]: will remove it

File: .github/workflows/main.yml (hunk 1 above)
Comment (gurdeep111994) [reply .github/workflows/main.yml]: ok

Comment (gurdeep111994) [reply .github/workflows/main.yml]: @chirgjn will you please check the given link?
https://www.split.io/blog/node-js-continuous-deployment-jenkins-heroku/

**akhileshns/heroku-deploy@v3.12.12 action created just for deploying to Heroku**

**Review by chirgjn — APPROVED** [review]

**Pull request merged**